### PR TITLE
Add tests to check the consistency of the default arguments between LightGBM and Optuna's LightGBM wrapper classes

### LINKED ISSUE #3037: Add test to ensure LightGBMTuner's default arguments and LightGBM's arguments consistent
## Motivation

Users suppose that LightGBMTuners follow the original methods' default arguments. To automatically detect the updates of the original methods, it would be great to add a test to check the Optuna's method are consistent with the original one.

## Description

Please add the test that ensures the default argument values are the same as the LightGBM's corresponding value to [`lightgbm_tuner_tests/test_optimize.py`](https://github.com/optuna/optuna/blob/master/tests/integration_tests/lightgbm_tuner_tests/test_optimize.py). More concretely, the following combination 

- [`LightGBMTuner`](https://optuna.readthedocs.io/en/stable/reference/generated/optuna.integration.lightgbm.LightGBMTuner.html#optuna-integration-lightgbm-lightgbmtuner) and [`lightgbm.train`](https://lightgbm.readthedocs.io/en/latest/pythonapi/lightgbm.train.html)
- [`LightGBMTunerCV`](https://optuna.readthedocs.io/en/stable/reference/generated/optuna.integration.lightgbm.LightGBMTunerCV.html#optuna.integration.lightgbm.LightGBMTunerCV) and [`lightgbm.cv`](https://lightgbm.readthedocs.io/en/latest/pythonapi/lightgbm.cv.html)

### PR DESCRIPTION
**Pull request by abatomunkuev**
Closes #3037  

## Motivation

Added unit tests to automatically detect the update of the original methods. This test ensures that Optuna's methods are consistent with the original ones. 

## Description of the changes

- Added a test default arguments in a class TestLightGBMTuner.  Provided an appropriate error message in assert statement. 
- Added a test default arguments in a class TestLightGBMTunerCV. Provided an appropriate error message in assert statement.
- Formatted the code

### CODE REVIEW

File: tests/integration_tests/lightgbm_tuner_tests/test_optimize.py [hunk 1]
```diff
@@ -360,6 +362,17 @@ def test__parse_args_wrapper_args(self) -> None:
...
+        for lgbmtuner_arg, lgbmtuner_val in signature(LightGBMTuner).parameters.items():
+            for lgbm_arg, lgbm_val in signature(lightgbm.train).parameters.items():
+                if lgbmtuner_arg == lgbm_arg:
+                    assert (
+                        lgbmtuner_val.default == lgbm_val.default
+                    ), (
+                        f"LightGBMTuner '{lgbmtuner_arg}' default "
+                        "argument is not consistent with original "
+                        "LightGBM 'train' method default argument."
+                    )
```
Comment (nzw0301) [inline tests/integration_tests/lightgbm_tuner_tests/test_optimize.py]: The code contains redundant logic. How about this? 
```suggestion
        tuner_kwargs = signature(optuna.integration.LightGBMTuner).parameters

        for lgb_train_arg, lgb_train_val in signature(lightgbm.train).parameters.items():

            if lgb_train_arg in tuner_kwargs:
                assert (
                    tuner_kwargs[lgb_train_arg].default == lgb_train_val.default
                ), (
                    f"LightGBMTuner '{lgb_train_arg}' default "
                    "argument is not consistent with original "
                    "LightGBM 'train' method default argument."
                )

```

File: tests/integration_tests/lightgbm_tuner_tests/test_optimize.py [hunk 2]
```diff
@@ -823,6 +836,18 @@ def test_with_minimum_required_args(self) -> None:
         assert "num_boost_round" not in runner.auto_options
         assert runner.lgbm_kwargs["num_boost_round"] == 5
 
+    def test_default_arguments(self) -> None:
```
Comment (nzw0301) [inline tests/integration_tests/lightgbm_tuner_tests/test_optimize.py:838]: Could you update this test as well?

**Review by nzw0301 — CHANGES_REQUESTED** [review]
Hi, thank you for sending your PR! 

Could you update this pull request by following my comment?

File: tests/integration_tests/lightgbm_tuner_tests/test_optimize.py (hunk 1 above)
Comment (abatomunkuev) [reply tests/integration_tests/lightgbm_tuner_tests/test_optimize.py]: Thanks for feedback. I am going to update the code.

**Comment by nzw0301** [comment]
Let me rename this PR for another reviewer and a release note.

**Comment by abatomunkuev** [comment]
@nzw0301 I have updated the code reflecting your feedback.

**Comment by nzw0301** [comment]
Thanks! As shown in tests for integrations, some parameters are not consistent, so we need to fix them.

I'll give concrete feedback later.

**Comment by nzw0301** [comment]
To fix one of the inconsistencies, could you

- replace `return_cvbooster: Optional[bool] = None,` with `return_cvbooster: bool = False,` at [this line](https://github.com/optuna/optuna/blob/master/optuna/integration/_lightgbm_tuner/optimize.py#L993), and
- remove these lines: https://github.com/optuna/optuna/blob/ea8c3c0a520b6d4e75230092b39d572ed6c8dc40/optuna/integration/_lightgbm_tuner/optimize.py#L1026-L1028

**Comment by nzw0301** [comment]
The other inconsistent default value, `num_boost_round`, might be the intended. Please give us a little bit more time to decide how to deal with this. Thanks!

**Comment by abatomunkuev** [comment]
@nzw0301  Hello! any news about `num_boost_round` parameter?

**Comment by nzw0301** [comment]
I'm afraid not :(

File: tests/integration_tests/lightgbm_tuner_tests/test_optimize.py [hunk 3]
```diff
@@ -360,6 +362,16 @@ def test__parse_args_wrapper_args(self) -> None:
...
+        tuner_kwargs = signature(LightGBMTuner).parameters
+        for lgb_train_arg, lgb_train_val in signature(lightgbm.train).parameters.items():
+            if lgb_train_arg in tuner_kwargs:
+                assert tuner_kwargs[lgb_train_arg].default == lgb_train_val.default, (
+                    f"LightGBMTuner '{lgb_train_arg}' default "
+                    "argument is not consistent with original "
+                    "LightGBM 'train' method default argument."
+                )
+
```
Comment (toshihikoyanase) [inline tests/integration_tests/lightgbm_tuner_tests/test_optimize.py:374]: I talked with some committers including @g-votte @hvy and @Crissman, and we came up with the approach to remove the hard-coded default values.

`LightGBMTuner` and `LightGBMTunerCV` have two types of arguments:
- Arguments with intentionally altered default values like `num_boost_round`: They should be hard-coded.
- Arguments with the same default values with the original LightGBM: we might be able to remove hard-coded default values and just pass through the arguments to the LightGBM functions.

If we can remove the hard-coded default values, we do not need this test case. So, how about focusing on the default value change of `return_cvbooster` and removing the new test cases in this PR?

Comment (abatomunkuev) [reply tests/integration_tests/lightgbm_tuner_tests/test_optimize.py:374]: Sure! I will remove the tests. 

Should I leave this: https://github.com/optuna/optuna/pull/3070#issuecomment-955964080 ?

Comment (nzw0301) [reply tests/integration_tests/lightgbm_tuner_tests/test_optimize.py:374]: Yes!

**Comment by abatomunkuev** [comment]
@nzw0301  Done!

**Review by nzw0301 — APPROVED** [review]
Brilliant!

**Review by HideakiImamura — APPROVED** [review]
LGTM!

**Comment by nzw0301** [comment]
Let me rename the PR title for the release note.

**Pull request merged**